azurelinuxagent/ga/update.py

```diff
@@ -728,6 +730,36 @@ def _write_pid_file(self):
 
         return pid_files, pid_file
 
+    @staticmethod
+    def _add_sym_link_for_ub20_04():
+        # Function to create a symlink for the python interpreter that the agent is running by
+        agent_dir_python_symlink_path = get_python_symlink_path_if_ubuntu_20_04_image()
+        if agent_dir_python_symlink_path is None:
+            # Not an Ubuntu 20.04 machine, skipping creating a symlink
+            return
+
+        add_event(
+            AGENT_NAME,
+            op=WALAEventOperation.Ubuntu2004_Image,
+            version=CURRENT_VERSION,
+            is_success=True,
+            message="Agent running on Ubuntu20.04, setting a python symlink in agent directory")
+
+        if os.path.islink(agent_dir_python_symlink_path):
```

Comment:
path.exists instead of path.islink?

Reply:
Initially I read that path.islink checks if its a symlink but wont return False if its broken but path.exists would even return false if the link is broken which is awesome! I'll replace it with path.exists